tests/rpmpkgfmt.at

```diff
@@ -20,6 +20,41 @@ rpm -qp --qf "%{rpmformat}\n" ${RPMTEST}/build/RPMS/4/noarch/attrtest-1.0-1.noar
 [4
 ],
 [])
+
+RPMTEST_CHECK([[
+sid=$(rpm -qp --qf "%{pkgid}\n" ${RPMTEST}/build/SRPMS/attrtest-1.0-1.src.rpm)
+pid=$(rpm -qp --qf "%{pkgid}\n" ${RPMTEST}/build/RPMS/4/noarch/attrtest-1.0-1.noarch.rpm)
+spid=$(rpm -qp --qf "%{sourcepkgid}\n" ${RPMTEST}/build/RPMS/4/noarch/attrtest-1.0-1.*.rpm)
+hid=$(rpm -qp --qf "%{hdrid}\n" ${RPMTEST}/build/RPMS/4/noarch/attrtest-1.0-1.noarch.rpm)
+shid=$(rpm -qp --qf "%{hdrid}\n" ${RPMTEST}/build/SRPMS/attrtest-1.0-1.src.rpm)
+
+# only validate the general characteristics here to avoid pinning
+echo -n ${sid} | wc -c
+echo -n ${pid} | wc -c
+echo -n ${spid} | wc -c
+echo -n ${hid} | wc -c
+echo -n ${shid} | wc -c
+[ "${sid}" = "${spid}" ] || exit 1
+[ "${sid}" != "${pid}" ] || exit 1
+[ "${pid}" != "${sid}" ] || exit 1
```

Comment:
This is a dupe of the previous check, just flipped :smile:

Reply:
Oh. I think the idea was to compare to spid but there's been a lossy transmission :sweat_smile:
Good spotting, will fix.